aggsender/aggsender.go

```diff
@@ -416,24 +428,120 @@ func (a *AggSender) sendCertificate(ctx context.Context) (*agglayertypes.Certifi
 	return certificate, nil
 }
 
-// callValidator calls the validator to validate the certificate
-func (a *AggSender) callValidator(
+// pollValidators committee signers to validate the certificate and gather signatures
+func (a *AggSender) pollValidators(
 	ctx context.Context,
+	validators []types.CertificateValidateAndSigner,
+	signaturesThreshold uint32,
 	certificate *agglayertypes.Certificate,
-	lastL2BlockInCert uint64) ([]byte, error) {
-	if a.validator == nil {
+	lastL2BlockInCert uint64,
+) ([][]byte, error) {
+	if len(validators) == 0 {
+		a.log.Warnf("skipping certificate validation, because there are no validators configured")
+
 		return nil, nil
 	}
+
 	a.log.Infof("delegating certificate validation: %s", certificate.Brief())
-	validatorSignature, err := a.validator.ValidateAndSignCertificate(ctx, certificate, lastL2BlockInCert)
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	sigCh := make(chan []byte, len(validators))
+	errCh := make(chan error, len(validators))
+
+	g, gctx := errgroup.WithContext(ctx)
+
+	for _, v := range validators {
+		g.Go(func() error {
+			status, err := v.HealthCheck(ctx)
+			if err != nil {
+				a.log.Warnf("error checking validator health (URL=%s): %v", v.URL(), err)
+			}
+
+			if !status.IsHealthy() {
+				a.log.Warnf("validator status (URL=%s) is not healthy: %s", v.URL(), status.String())
```

Comment:
Maybe return an error here?

Reply:
The idea was to just log and skip the unhealthy validators. Seems fine by me, but unsure if you are concerned about something particular?

Reply:
Ok, I think we are fine to log that it can't be reached, and it won't have a signature, so it doesn't matter if we return an error. Let's leave it like this.